### PR TITLE
[Models] Lfm2-VL Architecture

### PR DESCRIPTION
**Pull request by paulpak58**
## Purpose

LFM2-VL Implementation

## Test Result

---
<details>
<summary> Essential Elements of an Effective PR Description Checklist </summary>

- [ ] The purpose of the PR, such as "Fix some issue (link existing issues this PR will resolve)".
- [ ] The test plan, such as providing test command.
- [ ] The test results, such as pasting the results comparison before and after, or e2e results
- [ ] (Optional) The necessary documentation update, such as updating `supported_models.md` and `examples` for a new model.
- [ ] (Optional) Release notes update. If your change is user facing, please update the release notes draft in the [Google Doc](https://docs.google.com/document/d/1YyVqrgX4gHTtrstbq8oWUImOyPCKSGnJ7xtTpmXzlRs/edit?tab=t.0).
</details>

### CODE REVIEW

File: vllm/model_executor/models/lfm2_vl.py [hunk 1]
```diff
@@ -0,0 +1,732 @@
...
+        self.multimodal_config = multimodal_config
+        self.use_data_parallel = multimodal_config.mm_encoder_tp_mode == "data"
+
+        if vision_config.model_type == "siglip2_vision_model":
+            from vllm.model_executor.models.siglip2 import Siglip2Model
+
+            self.vision_tower = Siglip2Model(
+                config=vision_config,
+                quant_config=quant_config,
+                prefix=f"{prefix}.vit",
```
Comment (DarkLight1337) [inline vllm/model_executor/models/lfm2_vl.py]: Looks like a typo in the prefix

File: vllm/model_executor/models/siglip2.py [hunk 2]
```diff
@@ -0,0 +1,421 @@
...
+            quant_config=quant_config,
+            prefix=f"{prefix}.qkv_proj",
+        )
+        self.out_proj = RowParallelLinear(
+            input_size=self.embed_dim,
+            output_size=self.embed_dim,
+            quant_config=quant_config,
+            prefix=f"{prefix}.out_proj",
+        )
+        self.attn = MultiHeadAttention(
```
Comment (DarkLight1337) [inline vllm/model_executor/models/siglip2.py:173]: cc @Isotr0py @tjtanaa why does the existing SigLIP implementation not use `MultiHeadAttention` yet?

File: vllm/model_executor/models/siglip2.py [hunk 3]
```diff
@@ -0,0 +1,421 @@
...
+        )  # batch_size, q_len, 3 * num_heads_per_partition * head_dim
+        query_states, key_states, value_states = qkv.chunk(3, dim=-1)
+
+        # Use unified MultiHeadAttention implementation
+        out = self.attn(query_states, key_states, value_states)
+        attn_output, _ = self.out_proj(out)
+        return attn_output
+
+
+class Siglip2MLP(nn.Module):
```
Comment (DarkLight1337) [inline vllm/model_executor/models/siglip2.py]: For the other layers, you can import directly from siglip2 file if they are the same

File: vllm/model_executor/models/siglip2.py (hunk 2 above)
Comment (Isotr0py) [reply vllm/model_executor/models/siglip2.py:173]: Do you mean `siglip2navit`? Because it needs `cu_seq_lens` to build attention mask, which hasn't been integrated into `MultiHeadAttention`.

We will consolidate it with `MultiHeadAttention` after the vision attention refactoring (waiting https://github.com/vllm-project/vllm/pull/27919)

Comment (DarkLight1337) [reply vllm/model_executor/models/siglip2.py:173]: Yes

**Comment by DarkLight1337** [comment]
Closing as superseded by #31758
